### PR TITLE
Support config file

### PR DESCRIPTION
**Pull request by cffls**
This change allows users to 
* builder new docker images from a local path or a remote url.
* provide a configuration file to CLI so they don't need to enter options interactively.

### CODE REVIEW

File: src/setup/helper.js [hunk 1]
```diff
@@ -40,6 +40,15 @@ export async function getDefaultBranch(options = {}) {
 
   const questions = []
 
+  if (!options.borContext) {
```
Comment (temaniarpit27) [inline src/setup/helper.js:43]: we already ask user the bor docker tag
we can just ask him enter bor docker tag or branch
if docker tag exists - pull from that
if docker tag doesn't exist - we already clone the repo in code folder
we can just do a docker build there and use that image

in remote case
we can just locally check out that branch/tag and build binary

don't want to ask a lot of things from user as it becomes confusing for him

Comment (cffls) [reply src/setup/helper.js:43]: This is a good suggestion, thanks! Will try to use existing parameters.

Comment (cffls) [reply src/setup/helper.js:43]: Thought about this again. There are three potential issues if we only ask for a branch name:
1. User can create their own fork repo, but use the same release tag, e.g. repo: `userA/heimdall`, branch: `v0.2.9`. In this case, if we search for tag `v0.2.9` in docker hub, we will find it, but it is not necessarily the image that a user actually wants to use. 
2. Currently, the code clone function assumes the code is clonable from a url, and there is an existing branch under that repo. However, there is limitation to it. For example, we can't pull code from a pull request or from a git tag with current implementation. 
3. We can't build images from source code that is available only locally on users' machines. This means users need to push a commit to a remote repo every time when they want to test out a new change, which is not efficient.

Three challenges above could be solved if we provide a way for users to directly specify a docker build context, which allows images to be pulled and built from a Github repo with specific branch or tag, a local directory, or even a tarball file, see details [here](https://docs.docker.com/engine/reference/commandline/build/#git-repositories). In general, I think it is a good idea to separate docker build configuration from Git branch configuration. Combining them can cause confusions.

To address your concern of the increased complexity, I added a condition that will only ask users for docker context if the devnet type is docker. Let me know if you agree with this approach or if you have better ideas. Thanks!

Comment (cffls) [reply src/setup/helper.js:43]: As discussed with @temaniarpit27 offline, we will not add more questions in the helper, but instead, add a feature to support reading a config file, where we can add new flags and configurations to it.

File: src/setup/devnet/index.js [hunk 2]
```diff
@@ -498,7 +498,8 @@ async function setupDevnet(config) {
   devnet.genesis = new Genesis(config, { repositoryBranch: 'master' })
 
   const tasks = await devnet.getTasks()
-  await tasks.run()
```
Comment (temaniarpit27) [inline src/setup/devnet/index.js:501]: removed?

**Review by temaniarpit27 — CHANGES_REQUESTED** [review]

File: configs/README.md [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+```
+
+## Option details with examples
+
+```yaml
+# The root target directory where everything related to the network will be created
+targetDirectory: path/to/devnet 
+
+# Name of the directory where the code of bor, heimdall, and contracts will be cloned into
+codeDirectory: code
```
[the remark this answers is not in the document]
Comment (cffls) [reply configs/README.md:18]: Will remove this from this config.

File: configs/README.md [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+# URL to Ethereum RPC
+ethURL: http://ganache:9545
+
+# Devnet type, choose from [docker, remote]
+devnetType: docker
+
+# IPs of hosts where bor will run. Only effective when devnetType is remote.
+devnetBorHosts:
+  - 172.20.1.100
```
[the remark this answers is not in the document]
Comment (cffls) [reply configs/README.md:69]: In remote case, bor will directly point to `http://localhost:1317`. In docker case, these two parameters will not be required as they are automatically generated.

File: src/setup/devnet/index.js (hunk 2 above)
Comment (cffls) [reply src/setup/devnet/index.js:501]: Nice catch!

**Review by temaniarpit27 — APPROVED** [review]

**Comment by temaniarpit27** [comment]
LGTM
Just make sure that the flows are tested before merging

**Comment by cffls** [comment]
Tested both docker and remote flows.

**Pull request merged**